solr/solrj/src/test/org/apache/solr/common/cloud/TestZkConfigSetService.java

```diff
@@ -16,45 +16,63 @@
  */
 package org.apache.solr.common.cloud;
 
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+
 import com.google.common.base.Throwables;
+import org.apache.solr.SolrJettyTestBase;
 import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.cloud.AbstractZkTestCase;
 import org.apache.solr.cloud.ZkConfigSetService;
 import org.apache.solr.cloud.ZkTestServer;
 import org.apache.solr.core.ConfigSetService;
+import org.apache.solr.core.CoreContainer;
 import org.apache.zookeeper.KeeperException;
 import org.apache.zookeeper.ZooDefs;
 import org.apache.zookeeper.data.ACL;
 import org.apache.zookeeper.data.Id;
 import org.apache.zookeeper.server.auth.DigestAuthenticationProvider;
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
 import org.junit.Test;
 
-import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.security.NoSuchAlgorithmException;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-
 public class TestZkConfigSetService extends SolrTestCaseJ4 {
 
-  private static ZkTestServer zkServer;
+  private ZkTestServer zkServer;
+  protected Path zkDir;
+
+  private String solrHome;
 
-  @BeforeClass
-  public static void startZkServer() throws Exception {
-    zkServer = new ZkTestServer(createTempDir("zkData"));
+  private SolrZkClient zkClient;
+
+
+  @Override
+  public void setUp() throws Exception {
```

Comment:
The added setup & teardown is only used by one test, thus doing wasted stuff for the other tests.  You can move it to testBootstrapConf

Reply:
zkServer, zkClient are used by other tests in the class and they are set up/tear down in these methods;

Reply:
zkServer, zkClient are used by other test methods in the class and zkServer, zkClient are set up in these methods

Reply:
As we discussed in person, I'm saying the setup/teardown logic should be _inlined_ into where you are moving the test, basically.